cis194/week1/robert-chang/Homework01/src/Homework01.hs

```diff
@@ -0,0 +1,21 @@
+module Homework01
+    (
+      toDigits,
+      toDigitsRev,
+      doubleEveryOther, 
+      sumDigits
+    ) where
+
+toDigits :: Integer -> [Integer]
+toDigits num
+  | 0 >= num  = []
+  | otherwise = toDigits (num `div` 10) ++ [num `mod` 10]
+
+toDigitsRev :: Integer -> [Integer]
+toDigitsRev = reverse . toDigits
+
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther xs = reverse (zipWith ($) (cycle [id, (*2)]) (reverse xs))
```

Comment:
Since you used `$` here, which is not mentioned in the class, I feel like I can improve this with `.`.
```Haskell
doubleEveryOther = reverse 
                 . zipWith ($) (cycle [id, (*2)]) 
                 . reverse
```

Reply:
Thanks. This is neat!